### PR TITLE
[OLS-620] Use all fields from provider secret from the user

### PR DESCRIPTION
**Pull request by xrajesh**
## Description
Updated the secret mount path  not to include the filename , so that we can support all secret configurations.
Added azure_openai_config specific confing .

## Type of change

- [] Refactor
- [] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ x] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue [OLS-620](https://issues.redhat.com/browse/OLS-620)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: internal/controller/types.go [hunk 1]
```diff
@@ -35,15 +35,24 @@ type ProviderConfig struct {
...
 	Type string `json:"type,omitempty"`
-	// Azure deployment name
-	AzureDeploymentName string `json:"deployment_name,omitempty"`
 	// Watsonx Project ID
 	WatsonProjectID string `json:"project_id,omitempty"`
+	// Azure OpenAI Config
+	AzureOpenAIConfig *AzureOpenAIConfig `json:"azure_openai_config,omitempty"`
+}
+
+type AzureOpenAIConfig struct {
```
Comment (raptorsun) [inline internal/controller/types.go:49]: Are other fields listed in [the JIRA ticket](https://issues.redhat.com/browse/OLS-620) in the azureconfig section not neccessary?
```yaml           
   azureconfig:
                url: http://someurl
                client_secret_path: <path_to_file>
                tenant_id_path: <path_to_tenant_id>
                client_id_path: <path_to_client_id>
                api_key_path: <path_to_api_key>  (alternative supported today)
```

**Comment by xrajesh** [comment]
@raptorsun updated the JIRA ticket with document which has the final version we need to support. PTAL. We do not have any  group account for Azure OpenAI . I am in the process of obtaining one. I tested with Erik's apitoken.

**Pull request merged**